google/gax/__init__.py

```diff
@@ -482,3 +489,160 @@ def __next__(self):
         if self._index >= len(self._current):
             self._current = None
         return resource
+
+
+def _from_any(pb_type, any_pb):
+    if not any_pb.Is(pb_type.DESCRIPTOR):
+        raise TypeError(
+            'Could not convert {} to {}'.format(
+                any_pb.__class__.__name__, pb_type.__name__))
+
+    return pb_type.FromString(any_pb.value)
+
+
+class ResultError(GaxError):
+    """Thrown when an operation completes with an error result.
+    """
+
+    @classmethod
+    def from_operation_error(cls, operation_error):
+        """Factory: construct instance from an operation error result.
+
+        Args:
+            operation_error (google.rpc.Status): the error result of the
+                operation.
+        """
+        return cls(operation_error.message, operation_error.code)
+
+    def __init__(self, msg, code=None):
+        """Constructor.
+
+        Args:
+            msg (string): describes the error that occurred.
+            code (int, optional): the status code, which should be an enum
+                value of google.rpc.Code.
+        """
+        super(ResultError, self).__init__(msg)
+        self.code = code
+
+    def __str__(self):
+        msg = super(ResultError, self).__str__()
+        return 'ResultError({}, with code {})'.format(msg, self.code)
+
+
+class OperationFuture(object):
+    """A Future which polls a service for completion via OperationsApi."""
+
+    def __init__(self, operation, client, result_type, metadata_type,
+                 call_options=None):
+        """Constructor.
+
+        Args:
+            operation (google.longrunning.Operation): the initial long-running
+                operation.
+            client (google.cloud.gapic.longrunning.operations_api.OperationsApi):
+                the client used to manage the long-running operation with an API
+                service.
+            result_type (type): the class type to be unpacked from the result.
+            metadata_type (type): the class type to be unpacked from the
+                metadata.
+            call_options (google.gax.CallOptions, optional): the call options
+                that are used when reloading the operation.
+        """
+        self._first_operation = operation
```

Comment:
I don't think `_first_operation` is a needed property. The `_last_operation` property can be used where this is used per my comment on the metadata method and since the operation name will not change.